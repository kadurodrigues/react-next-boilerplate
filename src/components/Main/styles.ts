import styled from 'styled-components'

export const Wrapper = styled.main`
  background-color: #d9dbe6;
  color: #fff;
  width: 100%;
  height: 100%;
  padding: 3rem;
  display: flex;
  flex-direction: column;
  text-align: center;
  align-items: center;
  justify-content: center;
`
export const Title = styled.h1`
  font-size: 2.5rem;
  color: #114eff;
`
