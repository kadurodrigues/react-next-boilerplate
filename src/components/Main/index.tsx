import * as S from './styles'

const Main = () => (
  <S.Wrapper>
    <S.Logo
      src="/img/logo.svg"
      alt="Atom image and React Avancado written beside"
    ></S.Logo>
    <S.Title>React Avan√ßado</S.Title>
    <S.Description>
      Typescript, ReatJS, Next JS and Styled Components
    </S.Description>
    <S.Illustration
      src="/img/hero-illustration.svg"
      alt="A programmer sitting in front of a computer"
    ></S.Illustration>
  </S.Wrapper>
)

export default Main
