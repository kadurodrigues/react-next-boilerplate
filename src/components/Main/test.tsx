import { render, screen } from '@testing-library/react'
import Main from './index'

describe('<Main />', () => {
  it('should render the heading', () => {
    const { container } = render(<Main />)
    const headingFound = screen.getByRole('heading', {
      name: /React AvanÃ§ado/i
    })

    expect(headingFound).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render with the correct background color', () => {
    const { container } = render(<Main />)
    expect(container.firstChild).toHaveStyle({ 'background-color': '#06092b' })
  })
})
