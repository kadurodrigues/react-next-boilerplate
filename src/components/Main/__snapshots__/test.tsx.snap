// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the heading 1`] = `
<main
  class="styles__Wrapper-sc-1cs6c8-0 iAwbZK"
>
  <img
    alt="Atom image and React Avancado written beside"
    class="styles__Logo-sc-1cs6c8-1 MCbTX"
    src="/img/logo.svg"
  />
  <h1
    class="styles__Title-sc-1cs6c8-2 VHROD"
  >
    React Avan√ßado
  </h1>
  <h2
    class="styles__Description-sc-1cs6c8-3 jpDtNf"
  >
    Typescript, ReatJS, Next JS and Styled Components
  </h2>
  <img
    alt="A programmer sitting in front of a computer"
    class="styles__Illustration-sc-1cs6c8-4 eNveha"
    src="/img/hero-illustration.svg"
  />
</main>
`;
